packages/mco/components/modals/app-manage-policies/helper/consistency-groups.tsx

```diff
@@ -0,0 +1,425 @@
+            <SelectList>
+              {FILTERS.map((filter) => (
+                <SelectOption key={filter} value={filter}>
+                  {displayFilterText[filter]}
+                </SelectOption>
+              ))}
+            </SelectList>
+          </Select>
+        </FlexItem>
+        <FlexItem grow={{ default: 'grow' }}>
+          <SearchInput
+            placeholder={t('Search by name')}
+            value={searchValue}
+            onChange={(_event, value) => handleSearchChange(value)}
+            onClear={() => handleSearchChange('')}
+          />
+        </FlexItem>
+      </Flex>
+
+      <GroupList
+        filteredGroups={filteredGroups}
+        searchValue={searchValue}
+        t={t}
+      />
+    </ModalBody>
+  );
+};
+
+export const extractConsistencyGroups = (
+  mcvs: Record<string, any>
+): ConsistencyGroupInfo[] => {
+  const groups: ConsistencyGroupInfo[] = [];
+
+  Object.values(mcvs).forEach((mcvData) => {
+    if (!mcvData.loaded || mcvData.loadError || !mcvData.data) return;
+
+    const mcv = mcvData.data;
+    const vrg = mcv?.status?.result as DRVolumeReplicationGroupKind;
+    if (!vrg) return;
+
+    const cgMap = new Map<
+      string,
+      {
+        type: string;
+        pvcs: string[];
+        namespace: string;
+        lastSyncTime: string;
+        synced: boolean;
+      }
+    >();
+    const protectedPVCs = vrg.status?.protectedPVCs || [];
+    protectedPVCs.forEach((pvc) => {
+      const cgLabel = Object.entries(pvc.labels || {}).find(
+        ([key]) => key === CONSISTENCY_GROUP_LABEL
+      );
+
+      if (!cgLabel) return;
+
+      const cgName = cgLabel[1] as string;
+      const cgType = getCGType(cgName);
+
+      const schedulingInterval = vrg.spec.async?.schedulingInterval;
+      const dataLastSyncedOn = pvc.lastSyncTime;
+      const healthStatus = getReplicationHealth(
+        dataLastSyncedOn || '',
+        schedulingInterval,
+        getReplicationType(schedulingInterval)
+      );
+      const isSynced =
+        healthStatus === VolumeReplicationHealth.HEALTHY ? true : false;
+      if (!cgMap.has(cgName)) {
+        cgMap.set(cgName, {
+          type: cgType,
+          pvcs: [],
+          namespace: pvc.namespace,
+          lastSyncTime: pvc.lastSyncTime,
+          synced: isSynced,
+        });
+      }
+
+      const group = cgMap.get(cgName);
+      group.pvcs.push(pvc.name);
+    });
+
+    cgMap.forEach((value, cgName) => {
+      groups.push({
+        name: cgName,
+        type: value.type as any,
+        namespace: value.namespace,
+        lastSyncTime: value.lastSyncTime,
+        pvcs: value.pvcs,
+        synced: value.synced,
+      });
+    });
+  });
+
+  return groups;
+};
+
+const getMCVName = (
+  resource: DRPlacementControlKind | DRPlacementControlType
+): string => `${getName(resource)}-${getNamespace(resource)}-vrg-mcv`;
+
+export const buildMCVResources = (
+  application?: DRPlacementControlKind,
+  drInfo?: DRInfoType
```

Comment:
```suggestion
   clusterName: string,
   mcvName: string,
```
if you pass only these two you dont even need if else.